docs/docs-components/data/types.js

```diff
@@ -12,3 +12,64 @@ export type DesignOverview = {|
   path: string,
   svg: SVGElement,
 |};
+
+type StatusType = 'notAvailable' | 'partial' | 'planned' | 'ready';
+
+/**
+ * All components should be in a single list. That list can be filtered by category/platform/etc where needed.
+ */
+
+type ComponentAccessibility = {|
+  a11yComprehension?: StatusType,
+  a11yNavigation?: StatusType,
+  a11yScreenreader?: StatusType,
+  a11yVisual?: StatusType,
+  summary?: StatusType,
+|};
+
+type ComponentCategory =
+  | 'Actions'
+  | 'Avatars'
+  | 'Building blocks'
+  | 'Controls'
+  | 'Data'
+  | 'Fields and forms'
+  | 'Help and guidance'
+  | 'Indicators'
+  | 'Loading'
+  | 'Messaging'
+  | 'Navigation'
+  | 'Overlays'
+  | 'Pins and imagery'
+  | 'Structure'
+  | 'Text'
+  | 'Utilities';
+
+type PlatformObjType<T> = {|
+  android?: T,
+  figma?: T,
+  ios?: T,
+  web?: T,
+|};
+
+export type ComponentData = {|
+  id: string, // This is essentially the internal name
+  name: PlatformObjType<string>,
+  visual: {|
+    svg: PlatformObjType<SVGElement>,
+    hasDarkBackground?: boolean,
+  |},
+  alias: PlatformObjType<$ReadOnlyArray<string>>,
+  description: PlatformObjType<string>,
+  category: PlatformObjType<$ReadOnlyArray<ComponentCategory>>,
+  status: PlatformObjType<{|
```

Comment:
The current data structure is very repetitive, particularly vis-a-vis mobile platforms. This cuts down on the repetition and ensures a single source of truth.